### PR TITLE
Allow disabling the configuration of devices by unprivileged users

### PR DESCRIPTION
**Pull request by jamilbk**
Fixes firezone/product#418

Failing because of https://github.com/firezone/wireguardex/pull/34

Screen Shot 2022-08-10 at 3 53 20 PM

### CODE REVIEW

File: apps/fz_http/lib/fz_http_web/live/device_live/new_form_component.ex [hunk 1]
```diff
@@ -12,6 +13,7 @@ defmodule FzHttpWeb.DeviceLive.NewFormComponent do
   def mount(socket) do
     {:ok,
      socket
+     |> assign(allowed_to_configure?())
```
Comment (princemaple) [inline apps/fz_http/lib/fz_http_web/live/device_live/new_form_component.ex]: Is this not used? I don't see a `@allow_unprivileged_device_configuration` in the templates.

**Review by princemaple — DISMISSED** [review]
Just one question.

**Comment by princemaple** [comment]
Oh and also the name of the setting. I'm sure you have just many as regrets and internal struggles, so I'm going to pretend nothing bothered me. 😉

File: apps/fz_http/lib/fz_http_web/live/device_live/new_form_component.ex (hunk 1 above)
Comment (jamilbk) [reply apps/fz_http/lib/fz_http_web/live/device_live/new_form_component.ex]: PR is updated :-).

**Comment by jamilbk** [comment]
> Oh and also the name of the setting. I'm sure you have just many as regrets and internal struggles, so I'm going to pretend nothing bothered me. 😉

Yeah. Not thrilled with the name. I think this eventually belongs in an RBAC implementation consisting of `roles`, `actions` and `resources` so we can ask things like

`can_{action}?(user, resource)`

from RBAC policy config stored in the DB. Users have requested this, esp for tying groups / roles to firewall rules.

File: apps/fz_http/lib/fz_http_web/live/device_live/new_form_component.ex (hunk 1 above)
Comment (princemaple) [reply apps/fz_http/lib/fz_http_web/live/device_live/new_form_component.ex]: Umm what is?

Comment (jamilbk) [reply apps/fz_http/lib/fz_http_web/live/device_live/new_form_component.ex:16]: https://github.com/firezone/firezone/pull/909/files#diff-2551b8124ba4b1f13a9b5b8601ebbb51fa827bc7bccf2125bbcb7a6df96b9733R90

Comment (jamilbk) [reply apps/fz_http/lib/fz_http_web/live/device_live/new_form_component.ex]: Welp, *now* it's updated :-)

Comment (jamilbk) [reply apps/fz_http/lib/fz_http_web/live/device_live/new_form_component.ex]: Remove the `assign`

**Review by princemaple — DISMISSED** [review]

**Pull request merged**